Cython/Utility/ObjectHandling.c

```diff
@@ -2074,13 +2074,13 @@ static CYTHON_INLINE PyObject* __Pyx_CallUnboundCMethod1(__Pyx_CachedCFunction*
         // Not using #ifdefs for PY_VERSION_HEX to avoid C compiler warnings about unused functions.
         if (flag == METH_O) {
             return (*(cfunc->func))(self, arg);
-        } else if ((PY_VERSION_HEX >= 0x030600B1) && flag == METH_FASTCALL) {
-            if ((PY_VERSION_HEX >= 0x030700A0)) {
+        } else if (__PYX_CONST_CONDITION(PY_VERSION_HEX >= 0x030600B1) && flag == METH_FASTCALL) {
```

Comment:
Is it time for a `__Pyx_MIN_PY_VERSION()` macro? (Just asking…)

Reply:
Hm I would like something along the lines of
```cpp
#define __Pyx_PY_VERSION_HEX(MAJOR_VERSION, MINOR_VERSION, MICRO_VERSION, RELEASE_LEVEL, RELEASE_SERIAL, ...) \
   ((MAJOR_VERSION << 24) | \
    (MINOR_VERSION << 16) | \
    (MICRO_VERSION <<  8) | \
    (RELEASE_LEVEL <<  4) | \
    (RELEASE_SERIAL << 0))

#define __Pyx_PY_VERSION_GT(...) \
    (PY_VERSION_HEX > __Pyx_PY_VERSION_HEX(__VA_ARGS__, 0, 0, 0, 0, 0))
```
Which could be used e.g. as
```
__Pyx_PY_VERSION_GT(3)
```
However this would not work on c89/C++98 :/

I think if these macros are used they should be used for all operators:
```
__Pyx_PY_VERSION_LT()
__Pyx_PY_VERSION_LE()
__Pyx_PY_VERSION_EQ()
__Pyx_PY_VERSION_NE()
__Pyx_PY_VERSION_GE()
__Pyx_PY_VERSION_GT()
```
However I do not know whether either of
```
__Pyx_PY_VERSION_GE(3, 6, 0, PY_RELEASE_LEVEL_BETA, 1)
```
is much of an improvement over
```
PY_VERSION_HEX >= 0x030600B1
```

Reply:
I think virtually all version comparisons that we have can reduced to `PY_VERSION_HEX >= …` (and the tiny rest can just stay as it is). I also think it's fine to stick with the hex number. C level features/changes/incompatibilities are always introduced in some patch level version of CPython, and the hex version is a good way to express that.

Reply:
Should it be
```
#define __Pyx_MIN_PY_VERSION(VERSION_HEX) (PY_VERSION_HEX >= VERSION_HEX)
```
which works everywhere, or
```
#define __Pyx_MIN_PY_VERSION(VERSION_HEX) __PYX_CONST_CONDITION(PY_VERSION_HEX >= VERSION_HEX)
```
which does not work inside #ifdefs, but would simplify the usage inside if statements.